services/core-api/app/api/mines/permits/permit_amendment/models/permit_amendment.py

```diff
@@ -47,6 +47,21 @@ class PermitAmendment(AuditMixin, Base):
         UUID(as_uuid=True), db.ForeignKey('now_application_identity.now_application_guid'))
     now_identity = db.relationship('NOWApplicationIdentity', lazy='select')
 
+    def soft_delete(self, is_force_delete=False):
+        if not is_force_delete and self.permit_amendment_type_code == 'OGP':
+            raise Exception(
```

Comment:
I think we should raise a `BadRequest` as opposed to an `Exception` which I think will throw an "Internal Server Error".